rbi/core/thread.rbi

```diff
@@ -583,7 +583,22 @@ class Thread < Object
   #  from prog.rb:2:in `new'
   #  from prog.rb:2
   # ```
-  def raise(*_); end
+  sig {void}
+  sig do
+    params(
+      arg0: T.any(T::Class[Exception], Exception, String),
+    )
+    .void
+  end
+  sig do
+    params(
+      arg0: T.any(T::Class[Exception], Exception),
+      arg1: T.untyped,
+      arg2: T.nilable(T::Array[String]),
+    )
+    .void
+  end
+  def raise(arg0=nil, arg1=nil, arg2=nil); end
```

Comment:
Is there any reason for `Thread#raise` and `Kernel#raise` to have different signatures?

I notice that

- `Kernel#raise` returns `T.noreturn`
- `Kernel#raise` accepts `T::Class[T.anything]`, which does not look right given this:

```
irb(main):001> class A; end
=> nil
irb(main):002> raise A
(irb):2:in `raise': exception class/object expected (TypeError)
        from (irb):2:in `<main>'
        from /Users/jez/.rbenv/versions/3.1.4/lib/ruby/gems/3.1.0/gems/irb-1.14.0/exe/irb:9:in `<top (required)>'
        from /Users/jez/.rbenv/versions/3.1.4/bin/irb:25:in `load'
        from /Users/jez/.rbenv/versions/3.1.4/bin/irb:25:in `<main>'
```

Reply:
I suspect `Kernel#raise` signature was from before `T::Class` existed.

The return value is intentionally different. `Thread#raise` does return.

Reply:
ah makes sense, thanks!